compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -316,102 +445,227 @@ Fully qualified name: {fqn}
+    if items.is_empty() {
+        return;
     }
+
+    let short_name = if ns.starts_with("Microsoft.Quantum") {
+        ns.as_ref()
+    } else {
+        ns.split('.')
+            .last()
+            .expect("Namespaces should have at least one part.")
+    };
+
+    let package_kind = items[0].package.clone();
+    let metadata = Metadata {
+        uid: format!("Qdk.{ns}-toc"),
+        title: format!("{ns} namespace"),
+        topic: "landing-page".to_string(),
+        kind: MetadataKind::TableOfContents,
+        package: package_kind,
+        namespace: ns.clone(),
+        name: "Overview".into(),
+        summary: format!("Table of contents for the Q# {short_name} namespace"),
+        signature: String::new(),
+    };
+
+    items.sort_by_key(|item| item.name.clone());
+    let content = items
+        .iter()
+        .map(|item| {
+            format!(
+                "| [{}](xref:Qdk.{}) | {} |",
+                item.name,
+                item.fully_qualified_name(),
+                item.summary
+            )
+        })
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let content = format!(
+        "# {ns}
+
+The {ns} namespace contains the following items:
+
+| Name | Description |
+|------|-------------|
+{content}
+",
+    );
+
+    let rc_met = Rc::from(metadata);
+    items.insert(0, rc_met.clone());
+
+    let file_name = Rc::from(format!("{ns}/index.md").as_str());
+    let file_content = Rc::from(content.as_str());
+    files.push((file_name, rc_met, file_content));
 }
 
-impl Display for Metadata {
-    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
-        let kind = match &self.kind {
-            MetadataKind::Function => "function",
-            MetadataKind::Operation => "operation",
-            MetadataKind::Udt => "udt",
-            MetadataKind::Export => "export",
-        };
-        write!(
-            f,
-            "---
-uid: {}
-title: {}
-ms.date: {{TIMESTAMP}}
-ms.topic: {}
-qsharp.kind: {}
-qsharp.package: {}
-qsharp.namespace: {}
-qsharp.name: {}
-qsharp.summary: \"{}\"
----",
-            self.uid,
-            self.title,
-            self.topic,
-            kind,
-            self.package,
-            self.namespace,
-            self.name,
-            self.summary
-        )
-    }
+fn generate_top_index(files: &mut FilesWithMetadata, toc: &mut ToC) {
+    let empty_ns: Rc<str> = Rc::from(String::new().as_str());
+    let metadata = Metadata {
+        uid: "Microsoft.Quantum.apiref-toc".to_string(),
+        title: "Q# standard libraries for the Microsoft Quantum Development Kit".to_string(),
+        topic: "landing-page".to_string(),
+        kind: MetadataKind::TableOfContents,
+        package: PackageKind::StandardLibrary,
+        namespace: empty_ns.clone(),
+        name: "Overview".into(),
+        summary:
+            "Table of contents for the Q# standard libraries for Microsoft Quantum Development Kit"
```

Comment:
I think it's called "Azure Quantum Development Kit"

Reply:
updated